libs/agno/agno/agent/agent.py

```diff
@@ -2196,6 +2196,17 @@ def _run_tool(
         if len(function_call_results) > 0:
             run_messages.messages.extend(function_call_results)
 
+    def _reject_tool_call(self, run_messages: RunMessages, tool: ToolExecution):
+        self.model = cast(Model, self.model)
+        function_call = self.model.get_function_call_to_run_from_tool_execution(tool, self._functions_for_model)
+        function_call.error = tool.confirmation_note or "Function call was rejected by the user"
+
+        function_call_result = self.model.create_function_call_result(
+            function_call=function_call,
+            success=False,
+        )
+        run_messages.messages.append(function_call_result)
```

Comment:
Good! This can later also be the place to handle the user's comment if instead of just rejecting the tool call we allow for them to provide text

Reply:
Well we do use the `tool.confirmation_note` here.